spyder/plugins/ipythonconsole.py

```diff
@@ -1464,11 +1471,13 @@ def _create_client_for_kernel(self, connection_file, hostname, sshkey,
                                    _("Could not open ssh tunnel. The "
                                      "error was:\n\n") + to_text_string(e))
                 return
-        kernel_client.start_channels()
 
         # Assign kernel manager and client to shellwidget
         client.shellwidget.kernel_client = kernel_client
         client.shellwidget.kernel_manager = kernel_manager
+        client.sig_spyder_kernel.connect(self.connect_shellwidget)
+        kernel_client.start_channels()
+        client.shellwidget.is_spyder_kernel()

```

Comment:
This three lines *only* only apply when `master_name is None` because we don't want to do it for clients connected to kernels already created by Spyder.